cmd/producer/main.go

```diff
@@ -82,57 +80,46 @@ if async is preferred, then write request to redis.
 if symnchronous is preferred, then proxy the request.
 */
 func checkHeaderAndServe(w http.ResponseWriter, r *http.Request) {
-	target := &url.URL{
-		Scheme:   "http",
-		Host:     r.Host,
-		RawQuery: r.URL.RawQuery,
+	// if request body exists, check that length doesn't exceed limit
+	requestSizeInt, err := strconv.Atoi(env.RequestSizeLimit)
```

Comment:
why is this even left as string and not parsed as int by envconfig?
This would remove this fatal below as well  :)

Reply:
great, thanks!